### PR TITLE
Minor improvements in admin and docker-compose

### PR DESCRIPTION
**Pull request by PabloCastellano**
* [admin] Don't show the cancel button when there are no operations
* [docker] Add docker-compose for Redis

### CODE REVIEW

File: openwisp_firmware_upgrader/admin.py [hunk 1]
```diff
@@ -102,9 +102,9 @@ def upgrade_selected(self, request, queryset):
             {
                 'title': title,
                 'related_device_fw': related_device_fw,
-                'related_count': len(related_device_fw),
+                'related_count': related_device_fw.count(),
                 'firmwareless_devices': firmwareless_devices,
-                'firmwareless_count': len(firmwareless_devices),
+                'firmwareless_count': firmwareless_devices.count(),
```
Comment (nemesisdesign) [inline openwisp_firmware_upgrader/admin.py:107]: both querysets would be evaluated anyway during template rendering to build the list of devices, hence the full query will be done anyway.
So with this change, we're adding 2 more queries.
Or am I missing something?

Comment (nemesisdesign) [reply openwisp_firmware_upgrader/admin.py:107]: As anticipated, I added an assertion for the number of queries in the tests to test whether my assumption was correct.

Comment (PabloCastellano) [reply openwisp_firmware_upgrader/admin.py]: Well spotted, I've removed that part

File: openwisp_firmware_upgrader/templates/admin/upgrade_selected_confirmation.html [hunk 2]
```diff
@@ -52,7 +52,9 @@ <h2>{% trans "Devices which the system has never upgraded yet" %} ({{ firmwarele
   <input type="submit" name="upgrade_related" value="{% trans "Upgrade only related devices" %}">
   {% endif %}
   {% endif %}
+  {% if related_count or firmwareless_count %}
```
Comment (nemesisdesign) [inline openwisp_firmware_upgrader/templates/admin/upgrade_selected_confirmation.html:55]: So if there's no devices, how do the users go back? Maybe we could still provide the button but change its anchor text to show something like:

```
No devices to upgrade

<link>Go back</link>
```

Comment (PabloCastellano) [reply openwisp_firmware_upgrader/templates/admin/upgrade_selected_confirmation.html:55]: In my case I just clicked on the back button of my browser but I like the idea of having an extra button saying "Go back"

**Pull request merged**